### ISSUE TITLE
export StatisticsDefinition available on IStatisticDefinition

### ISSUE THREAD

**Issue by subgan82** [Esri/arcgis-rest-js#645]
Currently we are using `arcgis-rest-js` for Charts project. Can we export `StatisticsDefinition` type available on [IStatisticDefinition](https://esri.github.io/arcgis-rest-js/api/feature-layer/IStatisticDefinition/)

**Comment by tomwayson** [comment]
@subgan82 I'm sorry I missed this issue until today. If something's missing, I'm glad to accept a PR that adds it, or at least point me to where the definition is.

**Comment by subgan82** [comment]
Thanks @tomwayson . It is defined [here](https://github.com/Esri/arcgis-rest-js/blob/master/packages/arcgis-rest-feature-layer/src/query.ts#L30)

In the charts specification we can [replace](https://devtopia.esri.com/WebGIS/Online-Charts/blob/master/packages/arcgis-charts-spec/src/web-chart.ts#L126) it with this type. 
We are using `arcgis-rest-js` types in WebCharts specification removed JSAPI dependency at the spec level.

cc: @maneetgoyal

**Comment by tomwayson** [comment]
So looking at both of those interface definitions, it looks like the only difference is that `StatisticDefinition` allows a `statisticType` of `'median'`.

My first question is, is that valid value for arcgis server? All the examples in the [query docs](https://developers.arcgis.com/rest/services-reference/query-feature-service-layer-.htm) show `"statisticType": "<count | sum | min | max | avg | stddev | var>",`.

If it is a valid value, then we should include that in the `IStatisticDefinition` expression.

If it's not a valid value for arcgis server but you still need it for your app, then I think you could have `StatisticsDefinition` extend `IStatisticDefinition` and just overwrite `statisticType`.

**Comment by maneetgoyal** [comment]
- `median` was added as a temporary measure. It is currently not a valid value. 
- Yes, as you suggested, we were planning to extend `IStatisticDefinition` and just overwrite `statisticType` for the time being.
- But `IStatisticDefinition` may still need some corrections because some new statistics are [supported now](https://devtopia.esri.com/WebGIS/arcgis-js-api/issues/24952#issuecomment-2036394) (”PERCENTILE_CONT" and “PERCENTILE_DISC”).

**Comment by tomwayson** [comment]
> we were planning to extend IStatisticDefinition and just overwrite statisticType for the time being

OK, so what, if any, changes to this library do you need at this time?

**Comment by ubatsukh** [comment]
@tomwayson, you may find [this issue](https://devtopia.esri.com/WebGIS/arcgis-rest-api-documentation/issues/248).

**Comment by tomwayson** [comment]
w/r/t the new statistic types, at first glance it looks like there's a new optional `statistcParameters` property too that maybe only gets used for percentiles(???). I'd gladly accept a PR that updates the types in this library accordingly.

**Comment by maneetgoyal** [comment]
Based on this [comment](https://devtopia.esri.com/WebGIS/arcgis-js-api/issues/24952#issuecomment-2036517), 3 changes are needed. @ubatsukh, can you please confirm?

1. In `ILayerDefinition`, addition of `supportsPercentileStatistics`:
```
advancedQueryCapabilities?: {
    ...
    ...
    supportsPercentileStatistics?: boolean;
  };
```

2. In `IStatisticDefinition`:

- Addition of 2 new types: discrete percentile and continuous percentile.

```ts
statisticType: "count" | "sum" | "min" | "max" | "avg" | "stddev" | "var" | "percentile_cont" | "percentile_disc";
```

- Addition of an optional `statisticParameter` property

```ts
statisticParameter?: {
  value: number;
  orderBy?: "asc" | "desc"
}
```

In the [issue](https://devtopia.esri.com/WebGIS/arcgis-js-api/issues/24952#issuecomment-2036394) tagged earlier, upper-case is used but upon testing lower-case works fine. So to be consistent with other names, suggesting using lower-case.

@tomwayson We can raise a PR upon confirmation. cc @pemberdom @kumarGayu

**Comment by jgravois** [comment]
I know it can be really convenient to link to internal issues accessible to Esri staff, but it makes the conversation pretty cryptic for those of us following along from home. 😄

**Comment by maneetgoyal** [comment]
@jgravois Since that content was in private repositories, I wasn't sure if I am at the liberty to copy those comments here. But if you believe it is perfectly okay, I can post that info here. In essence:
- 2 new percentile statistics (discrete and continuous) are now supported.
- They need `statisticParameter` parameter also to work with.
- `statisticParameter.value` represents the `n-th` percentile we want to calculate.
- `statisticParameter.orderBy` represents how the data should be sorted before calculating the percentile. Default is `"asc"`.
- If a layer supports the calculation of these new statistics, then `supportsPercentileStatistics` in `advancedQueryCapabilities` is `true`.

@tomwayson After going through the updated [documentation](https://developersdev.arcgis.com/rest/services-reference/query-feature-service-layer-.htm) (available internally only), the changes look correct. Can you please provide access to create a PR?

**Comment by tomwayson** [comment]
Thanks @maneetgoyal - no access required here, just [fork, pull, and PR](https://gist.github.com/Chaser324/ce0505fbed06b947d962).

**Comment by subgan82** [comment]
Thanks @maneetgoyal for the PR. Thanks @tomwayson .

**Comment by tomwayson** [comment]
Released as [2.8.0](https://github.com/Esri/arcgis-rest-js/releases/tag/v2.8.0). Thanks!
